node/node-core/src/eth/mod.rs

```diff
@@ -0,0 +1,263 @@
+
+    #[serde(rename = "eth_getTransactionCount")]
+    EthGetTransactionCount(Address, Option<BlockNumber>),
+
+    #[serde(rename = "eth_getBlockTransactionCountByHash")]
+    EthGetTransactionCountByHash(H256),
+
+    #[serde(rename = "eth_getBlockTransactionCountByNumber")]
+    EthGetTransactionCountByNumber(BlockNumber),
+
+    #[serde(rename = "eth_getUncleCountByBlockHash")]
+    EthGetUnclesCountByHash(H256),
+
+    #[serde(rename = "eth_getUncleCountByBlockNumber")]
+    EthGetUnclesCountByNumber(BlockNumber),
+
+    #[serde(rename = "eth_getCode")]
+    EthGetCodeAt(Address, Option<BlockNumber>),
+
+    #[serde(rename = "eth_sendTransaction", with = "sequence")]
+    EthSendTransaction(Box<EthTransactionRequest>),
+
+    #[serde(rename = "eth_sendRawTransaction", with = "sequence")]
+    EthSendRawTransaction(Bytes),
+
+    #[serde(rename = "eth_call")]
+    EthCall(CallRequest, #[serde(default)] Option<BlockNumber>),
+
+    #[serde(rename = "eth_estimateGas")]
+    EthEstimateGas(CallRequest, #[serde(default)] Option<BlockNumber>),
+
+    #[serde(rename = "eth_getTransactionByHash", with = "sequence")]
+    EthGetTransactionByHash(TxHash),
+
+    #[serde(rename = "eth_getTransactionByBlockHashAndIndex")]
+    EthGetTransactionByBlockHashAndIndex(TxHash, Index),
+
+    #[serde(rename = "eth_getTransactionByBlockNumberAndIndex")]
+    EthGetTransactionByBlockNumberAndIndex(BlockNumber, Index),
+
+    #[serde(rename = "eth_getTransactionReceipt", with = "sequence")]
+    EthGetTransactionReceipt(H256),
+
+    #[serde(rename = "eth_getUncleByBlockHashAndIndex", with = "sequence")]
+    EthGetUncleByBlockHashAndIndex(H256, Index),
+
+    #[serde(rename = "eth_getUncleByBlockNumberAndIndex", with = "sequence")]
+    EthGetUncleByBlockNumberAndIndex(BlockNumber, Index),
+
+    #[serde(rename = "eth_getLogs")]
+    EthGetLogs(Filter),
+
+    #[serde(rename = "eth_getWork")]
+    EthGetWork,
+
+    #[serde(rename = "eth_submitWork", with = "sequence")]
+    EthSubmitWork(H64, H256, H256),
+
+    #[serde(rename = "eth_submitHashrate", with = "sequence")]
+    EthSubmitHashRate(U256, H256),
+
+    #[serde(rename = "eth_feeHistory")]
+    EthFeeHistory(
+        #[serde(deserialize_with = "deserialize_number")] U256,
+        BlockNumber,
+        #[serde(default)] Vec<f64>,
+    ),
+}
+
+fn deserialize_number<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum Numeric {
+        U256(U256),
+        Num(u64),
+    }
+
+    let num = match Numeric::deserialize(deserializer)? {
+        Numeric::U256(n) => n,
+        Numeric::Num(n) => U256::from(n),
+    };
+
+    Ok(num)
+}
+
+/// Represents a payload followed by an optional `block` value
+/// serialized as sequence of length 1: `[value, block?]`
+#[derive(Clone, Debug, PartialEq)]
+pub struct WithBlockParameter<T> {
+    pub value: T,
+    pub block: Option<BlockNumber>,
+}
```

Comment:
sort of surprised serde derive doesn't handle this

Reply:
this was weird bit I've figured it out and this type is no longer needed